src/storage/src/mut_records.rs

```diff
@@ -136,7 +167,89 @@ impl MutFileRecords {
 
     #[allow(unused)]
     pub async fn flush(&mut self) -> Result<(), IoError> {
-        self.f_sink.flush().await
+        self.flush_count.fetch_add(1, Ordering::Relaxed);
+        let mut f_sink = self.f_sink.lock().await;
+        debug!("flush: count {:?}", self.flush_count);
+        f_sink.flush().await
+    }
+
+    #[allow(unused)]
+    pub fn flush_count(&self) -> u32 {
+        self.flush_count.load(Ordering::Relaxed)
+    }
+
+    fn delay_flush(&mut self, delay_millis: u32) -> Result<(), IoError> {
+        let delay_tgt = delay_millis as u64;
```

Comment:
This seems like to spawn task every flush which is expensive.  And also creating channel as well.  
It probably more efficient if timer loop is created only once.

Reply:
The delay flush task is only spawned on the first write, after that, the flush_time_tx option is Some(channel::Sender) and the task will continue to loop on the rx of that channel for the rest of the lifetime of the mut_record instance. The following calls to delay_flush skip the        "if self.flush_time_tx.is_none() {" and  update the time sent to the task - which lets the loop start a new timer delay, or reason about in-process delays.

Reply:
got it.  just need to ensure that spawned future will be safely terminate if record is dropped

Reply:
When the mut_records struct is dropped, the only transmit side of the channel, flush_time_tx, is closed. The delay flush task picks that up via the while(!rx.is_closed()) condition and  exits the loop, ending the task.  I've verified that happens via tracing the test_write_records_idle_delay test.